### PR TITLE
[DevTestLab] Enable Shared/Public/Private IP configuration setting on lab vm

### PR DESCRIPTION
**Pull request by vishrutshah**
- [x] Enable Shared/Public/Private IP configuration setting on lab vm
- [x] Update documentation as per https://github.com/vishrutshah/azure-cli/pull/8
- [x] Update discovery of owner object id through https://github.com/Azure/azure-cli/pull/2648 logic

### CODE REVIEW

**Comment by vishrutshah** [comment]
@derekbekoe @tjprescott I've pushed one more commit onto this PR to be able to work with "default group configuration". I've verified that it is leveraging the default configuration. Feel free to take a quick re-look at the PR. Thanks!

**Comment by tjprescott** [comment]
LGTM

File: src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py [hunk 1]
```diff
@@ -120,29 +126,59 @@ def _validate_network_parameters(namespace, formula=None):
...
-        # Determine value for disallow_public_ip_address based on subnet's
-        # use_public_ip_address_permission property
-        if lab_vnet.subnet_overrides[0].use_public_ip_address_permission == 'Allow':
-            namespace.disallow_public_ip_address = False
-        else:
-            namespace.disallow_public_ip_address = True
+    _validate_ip_configuration(namespace, lab_vnet)
+
+
+# pylint: disable=no-member
```
Comment (troydai) [inline src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py]: why does it need a `no-member` exempt?

Comment (troydai) [inline src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py]: If it is one particular line need this exemption, make this a inline comments.

**Comment by troydai** [comment]
Do you need @ericame to sign off?

**Comment by vishrutshah** [comment]
I've informed her about the review but she seems busy with some other custom issue in Dev Test Lab team. Let's wait until 4:30 pm. If she gets chance to look into that's great or we can continue.

File: src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py (hunk 1 above)
Comment (vishrutshah) [inline src/command_modules/azure-cli-lab/azure/cli/command_modules/lab/validators.py]: There are few lines that'd require this comment. All the lines that accessed `lab_vnet`, so i'd prefer to keep it on method if that's fine & doesn't have drawbacks from python perspective :)

**Comment by vishrutshah** [comment]
As per offline conversation with @ericame I'll open an issue to check for existence of the "inbountNatRule" for defaulting logic.

**Pull request merged**